plasmapy/formulary/lengths.py

```diff
@@ -225,12 +235,21 @@ def gyroradius(
 
     isfinite_T = np.isfinite(T)
     isfinite_Vperp = np.isfinite(Vperp)
+    isfinite_lorentzfactor = np.isfinite(lorentzfactor)
 
     # check 1: ensure either Vperp or T invalid, keeping in mind that
     # the underlying values of the astropy quantity may be numpy arrays
-    if np.any(np.logical_and(isfinite_Vperp, isfinite_T)):
-        raise ValueError(
-            "Must give Vperp or T, but not both, as arguments to gyroradius"
+    if np.any(np.logical_not(np.logical_xor(isfinite_Vperp, isfinite_T))):
```

Comment:
Do you think we could add tests for these cases? If this logic was incorrect before, I suspect the existing ones must not have shown that, which is strange...

Reply:
> Functions that accept [array_like](https://numpy.org/doc/stable/glossary.html#term-array_like) or [Quantity](https://docs.astropy.org/en/stable/api/astropy.units.Quantity.html#astropy.units.Quantity) inputs should accept and return [nan](https://numpy.org/doc/stable/reference/constants.html#numpy.nan) ([not a number](https://en.wikipedia.org/wiki/NaN)) values.

Upon looking at the style guide I think leaving the output as NaN may have been intended so I will revert it to that.